pkg/test/framework/features/README.md

```diff
@@ -0,0 +1,47 @@
+# Overview
+The features package defines values that are used to track feature coverage at eng.istio.io
+# Labeling a Test
+Adding a label to a test using the integration framework is easy.  Simply call the Feature() function on the Test Object, passing any relevant feature constants as parameters.  If you are testing a feature that does not have a constant defined, see the next section for how to define one.
+# Adding New Feature Constants
+For consistency, feature constants are generated from `features.yaml`.  Each entry in this file will be generated into a dot delimited feature label.  For instance:
+
```

Comment:
@therealmitchconnors i don’t understand this fully.

Am I calling “feature(‘my.namespaced.feature’)” ?
If so, I suggest that the test author should consult feature.yaml and write the correct string.
In the feature impl, you can panic if someone is trying to report on a feature without registering it in feature.yaml.

If typos become an issue, we can have a 6 char random uid in addition to the display name , but I don’t think we will need it.

Reply:
Let's say we want to use the 'foo.bar' label, which has been declared in features.yaml.  The call should look like `feature(features.foo_bar)`.  I like the idea of checking the values against the yaml, as go provides very poor tooling to enforce type safety on this sort of thing...

Reply:
I am arguing for `no code gen`, but we simply run test that will fail if you try to make a `feature(...)` call without declaring it first. I think that is simpler.

Reply:
+1 on no code gen

Reply:
Updated docs to reflect no codegen.

Reply:
I'm fine with starting with runtime checking only and using a structure like the one @mandarjog proposes. We can add compile-time checking later if this proves insufficient.